### PR TITLE
develop: ci fix, some last backwards compatibility removals

### PR DESCRIPTION
**Pull request by ryanio**
This PR fixes an issue in develop where a karma workaround was erroring in the hardhat e2e installation for our packages. I moved the fix to the blockchain karma config.

I also executed some final breaking change items in VM and Trie.

### CODE REVIEW

**Comment by ryanio** [comment]
I noticed a series of my commits missing from my prior PR #1815:

Screen Shot 2022-04-04 at 9 25 38 PM

Highlighted here:

Screen Shot 2022-04-04 at 9 26 15 PM

So I have cherry-picked them and added them back to this PR.

**Comment by holgerd77** [comment]
Rebased this via UI with a local test rebase before.

File: packages/vm/src/index.ts [hunk 1]
```diff
@@ -284,9 +283,7 @@ export default class VM extends AsyncEventEmitter {
     if (opts.stateManager) {
       this.stateManager = opts.stateManager
     } else {
-      const trie = new Trie()
```
Comment (holgerd77) [inline packages/vm/src/index.ts:287]: 👍

File: packages/trie/src/baseTrie.ts [hunk 2]
```diff
@@ -104,7 +104,7 @@ export class Trie {
    */
   async checkRoot(root: Buffer): Promise<boolean> {
     try {
-      const value = await this._lookupNode(root)
+      const value = await this.lookupNode(root)
```
Comment (holgerd77) [inline packages/trie/src/baseTrie.ts:107]: We didn't plan to do a breaking Trie release initially, but my previously somewhat stronger opinion on this is wayning, so we might just do along (and then also finally do the renaming).

Just wanted to mention here. We then eventually want to have one additional thought round or broader look on other breaking changes.

Comment (holgerd77) [reply packages/trie/src/baseTrie.ts:107]: So, some additional context: currently Trie is still listed in the "Future Release Planning" issue: https://github.com/ethereumjs/ethereumjs-monorepo/issues/1718. Should I move over? 🤔

File: packages/client/lib/service/fullethereumservice.ts [hunk 3]
```diff
@@ -190,9 +191,18 @@ export class FullEthereumService extends EthereumService {
...
+          receipts.map((r) =>
+            rlp.encode([
+              (r as PreByzantiumTxReceipt).stateRoot ?? (r as PostByzantiumTxReceipt).status,
+              bigIntToBuffer(r.gasUsed),
+              r.bitvector,
+              r.logs,
+            ])
+          )
+        )
+        receiptsSize += Buffer.byteLength(receiptsBuffer)
```
Comment (holgerd77) [inline packages/client/lib/service/fullethereumservice.ts:195]: Could you drop a short note what's happening here?

Does this fix some concrete issues? Can't oversee this on a first look.

**Review by holgerd77 — APPROVED** [review]
Nice PR 🙂, looks good! 👍

File: packages/trie/src/baseTrie.ts (hunk 2 above)
Comment (ryanio) [reply packages/trie/src/baseTrie.ts:107]: ah sorry didn't notice that.

yeah sure let's include these refactorings, i think now is the time to do it while we are preparing develop. not sure if it makes sense to wait any longer. we can assign some team members to the tasks and i can also dedicate some time this month to tackle them.

File: packages/client/lib/service/fullethereumservice.ts (hunk 3 above)
Comment (ryanio) [reply packages/client/lib/service/fullethereumservice.ts:195]: ah yeah, actually you commented on an old version of this code, if you look at the final version in this PR i made it much more simple by using our existing encodeReceipt function. this is simply keeping track of the size of the GetReceipts response, so we can respect capping at 10mib as the spec suggests.

**Pull request merged**